### PR TITLE
ci: use composite action for tmate session debugging

### PR DESCRIPTION
**Pull request by ideepika**
this will help deduplicate tmate debug code throughout.

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/tmate_debug/action.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+description: "Setup tmate session if the test fails"
+inputs:
+  use-tmate:
+    description: "boolean for enabling TMATE"
+    required: false
+  timeout-minutes:
+    description: "timeout tmate debug session"
+    required: false
+    default: '60'
+# ...name, description and inputs as above
```
Comment (subhamkrai) [inline .github/workflows/tmate_debug/action.yml:11]:
```suggestion
```
can we remove this line?

**Review by subhamkrai — CHANGES_REQUESTED** [review]

**Comment by ideepika** [comment]
logs for working case where tmate session should be created: https://github.com/rook/rook/actions/runs/4231680932/jobs/7350498692

**Review by subhamkrai — APPROVED** [review]
LGTM

**Review by galexrt — APPROVED** [review]

**Comment by galexrt** [comment]
@travisn Are you fine with merging the PR with the failing `multi-cluster-mirroring` test?

File: .github/workflows/tmate_debug/action.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    description: "boolean for enabling TMATE"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: consider debugging
+      shell: bash --noprofile --norc -eo pipefail -x {0}
+      run: |
+        # Enable tmate only in the Rook fork, where the USE_TMATE secret is set in the repo, or if the action is re-run
+        if [ "$GITHUB_REPOSITORY_OWNER" = "rook" ] || [ -n "${{ inputs.use-tmate }}" ] || [ "$GITHUB_RUN_ATTEMPT" -gt 1 ]; then
```
Comment (parth-gr) [inline .github/workflows/tmate_debug/action.yml:14]: Is the tmate session behaved the same as it was before for the failed tests

Comment (ideepika) [reply .github/workflows/tmate_debug/action.yml:14]: yes: https://github.com/rook/rook/actions/runs/4231680932/jobs/7350498692

Comment (galexrt) [reply .github/workflows/tmate_debug/action.yml:14]: Yes, it also ran for at roughly 60 minutes, see https://github.com/rook/rook/actions/runs/4231724283/jobs/7392465051

File: .github/workflows/push-build.yaml [hunk 3]
```diff
@@ -78,13 +78,7 @@ jobs:
...
-        run: |
-          # Enable tmate only in the Rook fork, where the USE_TMATE secret is set in the repo, or if the action is re-run
-          if [ "$GITHUB_REPOSITORY_OWNER" = "rook" ] || [ -n "${{ secrets.USE_TMATE }}" ] || [ "$GITHUB_RUN_ATTEMPT" -gt 1 ]; then
-            echo USE_TMATE=1 >> $GITHUB_ENV
-          fi
-
-      - name: set up tmate session for debugging
-        if: failure() && env.USE_TMATE
-        uses: mxschmitt/action-tmate@v3
         timeout-minutes: 60
```
Comment (parth-gr) [inline .github/workflows/push-build.yaml:81]: can you also include this in the test `timeout-minutes: 60` mostly it is common

**Review by parth-gr — CHANGES_REQUESTED** [review]

File: .github/workflows/push-build.yaml (hunk 3 above)
Comment (ideepika) [reply .github/workflows/push-build.yaml:81]: This is still to be implemented in composite action, they don't support timeout-minutes currently, i mentioned it in commit msg, once the support comes out, i will update the code for the same, thanks

Comment (parth-gr) [reply .github/workflows/push-build.yaml:81]: okay thanks

**Review by parth-gr — APPROVED** [review]

**Comment by travisn** [comment]
> @travisn Are you fine with merging the PR with the failing `multi-cluster-mirroring` test?

Yes, that one is failing consistently in master. #11744 is tracking it.

**Review by travisn — APPROVED** [review]

**Comment by galexrt** [comment]
Merging now. The only CI test running (again) right now is the `multi-cluster-mirroring` which is as pointed out by @travisn tracked, I had restarted it a few minutes before your comments.

**Pull request merged**